src/scripts/developer.ts

```diff
@@ -0,0 +1,98 @@
+import { validateDeveloper, generateId } from "./helpers/index.js";
+
+export function addDeveloper(developers: Developers, developer: Partial<Developer>) {
+	const isDevValidated = validateDeveloper(developer);
+
+	if (isDevValidated) return;
+
+	const validatedDev = {
+		...developer,
+		id: generateId(developers),
+	};
+	developers.push(validatedDev as Developer);
+
+	alert(`Developer ${developer.name} added`);
+}
+
+export function cloneDeveloper(developer: Developer, developers?: Developers): Developer {
+	const deepCopy: Developer = {
+		...JSON.parse(JSON.stringify(developer)),
+		id: developers !== undefined ? generateId(developers) : developer.id,
+	};
+
+	return deepCopy;
+}
+
+export function updateDeveloper(developer: Developer, updates: Partial<Developer>): Developer {
+	const clonedDev = cloneDeveloper(developer);
+	const { name, age, experience, isEmployed, projects, skills } = clonedDev;
```

Comment:
Add null check

Reply:
where i have to add null check in this

Reply:
Like this -> const { name, age, experience, isEmployed, projects, skills } = clonedDev || {};

Reply:
ok